### ISSUE TITLE
pubsub: subscriptions are acting differently

### ISSUE THREAD

**Issue by willie** [googleapis/google-cloud-go#6525]
**Client**

My code using cloud.google.com/go/pubsub v1.17.1

**Environment**

Ubuntu 18 in Docker on GKE

**Expected behavior**

Messages arrive really fast.

**Actual behavior**

Around 3 am Central on August 16, subscription messages no longer arrive after an initial burst on app startup. The Go program I wrote has been working in production since the end of December 2021, with some hiccups in the last few weeks (no code changes)

**Additional context**

Updating to latest clients did not help, so I reverted to what had been working prior to July/August.

cloud.google.com/go/pubsub v1.17.1

**Comment by hongalex** [comment]
Thanks for filing an issue. To be clear, did this issue start happening after migrating to a new version beyond v1.17.1? Is v1.17.1 the last known working version?

Also, can you describe what your backlog looks like like?

**Comment by willie** [comment]
Last known working version was 1.17.1. I tried the latest, but reverted when it made no change. The backlog is growing quickly.

<img width="617" alt="Screen Shot 2022-08-17 at 10 27 15 PM" src="https://user-images.githubusercontent.com/20665/185286558-75f969c7-8b14-45ba-a46f-c422c99d4461.png">

As you can see, it started at 3:00 am CDT on August 16.

<img width="953" alt="Screen Shot 2022-08-17 at 10 28 32 PM" src="https://user-images.githubusercontent.com/20665/185286777-fd59f29f-1f37-4a1c-b335-584b190d8b4b.png">

**Comment by willie** [comment]
Good news! It wasn't pubsub. My apologies: I feel stupid.

Backstory
We take log messages from pubsub, filter and massage the data, group the messages into batches for insert throughput into our elastic cluster with inserts going round robin into data nodes.

When this started happening, I reached out to our devops folks and asked if elastic was fine. They replied: the cluster is green, everything seems fine, plenty of disk.

And each time we ran our program it would insert messages into elastic for a little while and then die.

The last couple of days, I rewrote the code to use less pools of goroutines and channels so that it would be easier to debug. Today, while working on a trial deploy of this rewrite and things were acting exactly the same, the sys admin I was talking to said, "the 5 data nodes look happy". I replied, "the config says there are 6 data nodes". He replied, "oh it looks like node 5 is down". 🤦‍♂️ 

Here is what was happening: all the inserts into the other nodes got through fine. Inserts to the 5th data node weren't getting refused, weren't getting timed out (not reasonably), etc. The inserts were getting stuck and held for a long time. So long that eventually my elastic insert concurrency hit a limit and everything ground to a halt waiting for the elastic inserts to timeout or whatever.

Reboot the data node, let it rejoin the cluster and everything works again. I need to work on my HTTP connection pool and timeouts.

Again, my apologies for the false fire drill.
